quickwit/quickwit-config/src/source_config/mod.rs

```diff
@@ -135,15 +136,15 @@ impl SourceConfig {
     /// Returns `desired_num_pipelines` if it's a Kafka source, else 1.
     pub fn desired_num_pipelines(&self) -> usize {
         match &self.source_params {
-            SourceParams::Kafka(_) => self.desired_num_pipelines,
+            SourceParams::Kafka(_) | SourceParams::Pulsar(_) => self.desired_num_pipelines,
```

Comment:
What is the actual implication here with this on a source?

Does this prevent us from having multiple pulsar sources for different indexes per indexer, or is this for the specific source itself?

Reply:
> Does this prevent us from having multiple pulsar sources for different indexes per indexer, or is this for the specific source itself?

It's for the specific source. I'm testing pulsar on EKS and because of this, the control plane believes the pulsar source can only have 1  `desired_num_pipelines` and I can't test the distribution!!!!

Reply:
I hesitate to remove `pub fn desired_num_pipelines` and `pub fn max_num_pipelines_per_indexer`, I will open an issue on this.